currency_rate_update/models/res_currency_rate_provider.py

```diff
@@ -0,0 +1,299 @@
+# Copyright 2009-2016 Camptocamp
+# Copyright 2010 Akretion
+# Copyright 2019 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import logging
+from datetime import datetime, time
+from dateutil.relativedelta import relativedelta
+
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class ResCurrencyRateProvider(models.Model):
+    _name = 'res.currency.rate.provider'
+    _description = 'Currency Rates Provider'
+    _inherit = ['mail.thread']
+    _order = 'name'
+
+    company_id = fields.Many2one(
+        string='Company',
+        comodel_name='res.company',
+        required=True,
+        default=lambda self: self._default_company_id(),
+    )
+    active = fields.Boolean(
+        default=True,
+        track_visibility='onchange',
+    )
+    service = fields.Selection(
+        string='Source Service',
+        selection=[],
+        required=True,
+        track_visibility='onchange',
+    )
+    currency_ids = fields.Many2many(
+        string='Currencies',
+        comodel_name='res.currency',
+        column1='provider_id',
+        column2='currency_id',
+        help='Currencies to be updated by this provider',
+        domain=lambda self: self._domain_currency_ids(),
+        track_visibility='onchange',
+    )
+    name = fields.Char(
+        string='Name',
+        compute='_compute_name',
+        store=True,
+    )
+    interval_type = fields.Selection(
+        string='Units of scheduled update interval',
+        selection=[
+            ('days', 'Day(s)'),
+            ('weeks', 'Week(s)'),
+            ('months', 'Month(s)'),
+        ],
+        default='days',
+        required=True,
+        track_visibility='onchange',
+    )
+    interval_number = fields.Integer(
+        string='Scheduled update interval',
+        default=1,
+        required=True,
+        track_visibility='onchange',
+    )
+    update_schedule = fields.Char(
+        string='Update Schedule',
+        compute='_compute_update_schedule',
+    )
+    next_run = fields.Date(
+        string='Next scheduled update',
+        default=fields.Date.today(),
```

Comment:
You should just pass `fields.Date.today` to get as default "today's date" when the record is created. Currently you are calling it which will get evaluated when the field is instantiated instead.

Reply:
Good catch, fixed!